src/main/java/com/hillel/items_exchange/service/impl/ImageServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+package com.hillel.items_exchange.service.impl;
+
+import com.hillel.items_exchange.dao.ImageRepository;
+import com.hillel.items_exchange.dto.ImageDto;
+import com.hillel.items_exchange.model.Image;
+import com.hillel.items_exchange.model.Product;
+import com.hillel.items_exchange.service.ImageService;
+import com.hillel.items_exchange.service.ProductService;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.TypeToken;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.imageio.IIOImage;
+import javax.imageio.ImageIO;
+import javax.imageio.ImageWriteParam;
+import javax.imageio.ImageWriter;
+import javax.imageio.stream.ImageOutputStream;
+import java.awt.image.BufferedImage;
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class ImageServiceImpl implements ImageService {
+    private final ModelMapper modelMapper;
+    private final ImageRepository imageRepository;
+
+    @Override
+    public List<byte[]> getImagesResourceByProductId(long productId) {
+        return imageRepository.findByProductId(productId).stream()
+                .map(Image::getResource)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<ImageDto> getByProductId(long productId) {
+        return mapImagesToDto(imageRepository.findByProductId(productId));
+    }
+
+    private List<ImageDto> mapImagesToDto(Iterable<Image> images) {
+        return modelMapper.map(images, new TypeToken<List<ImageDto>>() {}.getType());
+    }
+
+    @Override
+    public List<byte[]> compressImages(List<MultipartFile> images) throws IOException {
+        List<byte[]> compressedImages = new ArrayList<>();
+        for (MultipartFile photo : images) {
+            compressedImages.add(compressImage(photo));
+        }
+        return compressedImages;
+    }
+
+    @Override
+    public byte[] compressImage(MultipartFile image) throws IOException {
+
+        String contentType = image.getContentType();
+        if (contentType == null || contentType.equals(MediaType.IMAGE_GIF_VALUE)) {
```

Comment:
I noticed you check in all content types of received images in Image controller at 66-68 lines, don't you? Why do you check content type again?
Also you don't compress gif files at all, am I right? Why?

Reply:
@lubchik-14 .gif files couldn't be compressed at all. I agree that it worth to check file types here as well - I'll do it.